File: src/main/scala/example1/Address.java
package example1;

import com.datastax.driver.mapping.annotations.Field;
import com.datastax.driver.mapping.annotations.UDT;

@UDT(keyspace = "ks", name = "address")
public class Address {

    private String street;

    @Field(name = "zip_code")
    private int zipCode;

    public Address() {
    }

    public Address(String street, int zipCode) {
        this.street = street;
        this.zipCode = zipCode;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public int getZipCode() {
        return zipCode;
    }

    public void setZipCode(int zipCode) {
        this.zipCode = zipCode;
    }

    @Override
    public String toString() {
        return "Address{" +
                "street='" + street + '\'' +
                ", zipCode=" + zipCode +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Address address = (Address) o;

        if (zipCode != address.zipCode) return false;
        return street != null ? street.equals(address.street) : address.street == null;
    }

    @Override
    public int hashCode() {
        int result = street != null ? street.hashCode() : 0;
        result = 31 * result + zipCode;
        return result;
    }
}

File: src/main/scala/example2/Address.scala
package example2

import com.datastax.driver.mapping.annotations.Field
import com.datastax.driver.mapping.annotations.UDT

import scala.annotation.meta.field
import scala.beans.BeanProperty

@UDT(keyspace = "ks", name = "address")
case class Address(
  @BeanProperty street: String,
  @BeanProperty @(Field @field)(name = "zip_code") zipCode: Integer
) {

  // default constructor with nulls are used for java example interoperability
  def this() = this(null, null)
}
